### PR TITLE
fix(chat): disable approve button and add error message if unpublished entity exists in approve required modal (Issue #1422)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

disable approve button and add error message if unpublished entity exists in approve required modal

Issues:

- #1422 

**UI changes**

https://www.figma.com/design/lwL5jneveZczAj5RLcLFH2/DIAL-Chat?node-id=10248-13602&t=E8mky5FqoANtijJF-4

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1749.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/Publish/HandlePublication.tsx [hunk 1]
```diff
@@ -269,6 +273,10 @@ export function HandlePublication({ publication }: Props) {
     ? publication.targetFolder.replace(/^[^/]+/, 'Organization')
     : '';
 
+  const invalidEntities = deletedEntities.filter((entity) =>
+    publication.resources.map((r) => r.reviewUrl).includes(entity.id),
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/HandlePublication.tsx]: I would suggest to change to
```suggestion
    publication.resources.some((r) => r.reviewUrl === entity.id),
```

File: apps/chat/src/types/common.ts [hunk 2]
```diff
@@ -26,7 +26,8 @@ export enum BackendResourceType {
 }
 
 export interface EntityPublicationInfo {
-  action: PublishActions;
+  action?: PublishActions;
+  isDeleted?: boolean;
```
Comment (IlyaBondar) [inline apps/chat/src/types/common.ts]: we already have `PublishActions.DELETE`.
Why do we need `isDeleted` here?

Comment (Alexander-Kezik) [reply apps/chat/src/types/common.ts]: it how I marked entities which have been unpublished. Renamed to `inNotExist`

**Review by IlyaBondar — APPROVED** [review]

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1749.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**